detector.go

```diff
@@ -195,32 +209,55 @@ func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupElem
 			return d.detectOrder(groupEl.OrderExtensions, done, group.Group[i+1:], true, wg)
 		}
 
-		// Lookup element in store.
+		// Lookup element in store.  <-- "the store" is the directory where all the buildpacks are.
 		var (
-			descriptor buildpack.Descriptor
-			err        error
+			descriptor   buildpack.Descriptor
+			bpDescriptor *buildpack.BpDescriptor
+			err          error
 		)
 		if groupEl.Kind() == buildpack.KindBuildpack {
-			descriptor, err = d.DirStore.LookupBp(groupEl.ID, groupEl.Version)
+			bpDescriptor, err = d.DirStore.LookupBp(groupEl.ID, groupEl.Version)
 			if err != nil {
 				return nil, nil, err
 			}
 
+			targetMatch := false
+			if d.AnalyzeMD.RunImage.Target.IsWildcard() {
```

Comment:
hmm. i think just semantic readability -- confessionally, in my heart i'm a bit of a ruby/python programmer who wishes that each line of code could be an (admittedly kind of awkward) english sentence, so this appeals to me.  But I see your point that i've pulled out a function with a name for some logic that i'm really only using in one place.  Feel free to say something about "stylistically historically in this codebase we prefer... " and I can  just inline it, but re-reading it, it still passes my own personal threshold at this moment  :)

Reply:
I was thinking about it from a external API builder. If we think a platform user may have the need to run this code, say in a custom `prepare` type of step, it would make it so they wouldn't have to know to check for wildcard explicitly. But maybe this file/validation isn't likely to be consumed externally anyway.

Reply:
ok yeah, i gotchu --  moved into the detector and lowercased.